lib/events/kafka_listener.rb

```diff
@@ -26,21 +26,42 @@ def subscribe
       Rails.logger.error(["Something is wrong with Kafka client: ", e.message, *e.backtrace].join($RS))
       retry
     end
-  
+
     private
-  
+
     def raw_process(event)
       Rails.logger.info("Kafka message #{event.message} received with payload: #{event.payload}")
-  
-      ActiveRecord::Base.connection_pool.with_connection do
-        process_event(event)
+
+      insights_headers = event.headers.slice('x-rh-identity', 'x-rh-insights-request-id')
+      if insights_headers.empty?
```

Comment:
@hsong-rh We need a comment here to indicate that some of the messages in Kafka don't have the headers and we need to generate the header from the message body. We also account for that fact that not all messages are going to have the alternate request_id and b64_identity keys. That is the format that is being used by the ingress service.